### ISSUE TITLE
Clarification Regarding Complaince with Google Play Policies

### ISSUE THREAD

**Issue by sakibanda** [SuddenH4X/awesome-app-rating#99]
Hello,

Recently my app was suspended and I see one of the reasons saying it cannot use "Techniques to evade app reviews are not allowed."

I am not sure whether this SDK is compliant with the Google Play Policy or not when using the function setRatingThreshold, please clarify.

**Comment by SuddenH4X** [comment]
Hi @sakibanda,

thanks for your message. It appears that Google has recently updated their policies regarding app reviews. One of the new rules is:

> Techniques to evade app reviews are not allowed.

(See [Deceptive Behavior](https://support.google.com/googleplay/android-developer/answer/9888077?hl=en))

This rule was announced in July, but there was a grace period until a few days ago. As a result, using the libraries dialog isn't compliant anymore. I will remove the custom dialog from this library in a future release and recommend using the in-app review feature instead.

**Comment by rohitjakhar** [comment]
But still i see many app like bumble, they show dialog to ask user to review their app on playstore.
How do they do it?

**Comment by sakibanda** [comment]
Yes, many apps are still asking to rate 5 stars but no action is taken by Google, those are popular apps with millions of users. But, they are targeting new apps without much users. Google responded saying I need to file a complaint if I have any problem with those apps.

**Comment by rohitjakhar** [comment]
I think your implementation may be wrong, if you show proper message for app rating so there is no issue.
If possible can you show me the UI?

**Comment by sakibanda** [comment]
Hi Rohit, No, i dont mean it in a wrong, I meant that apps which are affecting millions of users has to be reviewed carefully. I don't have any screenshots but I remeber seeing on popular apps, show text sounding "Rate us 5 Stars" , "Do you like using the app? Yes or No" and based on answer app shows play store rating. I don't think these are complaint with latest policy.

I will post screenshots when I come across next time.

**Comment by SuddenH4X** [comment]
I'm still uncertain about whether to remove the libraries dialog and recommend the in-app review feature instead. Have you had any new experiences with this? How did you resolve the issue, @sakibanda?

**Comment by sakibanda** [comment]
Hi @SuddenH4X , I am not sure what to do either, I am just simply using only Google InApp Rating on a safer side. I faced lot of trouble resolving this issue.

**Comment by suhasrekhu** [comment]
I think there is a confusion - Techniques to evade app reviews are not allowed.

(See [Deceptive Behavior](https://support.google.com/googleplay/android-developer/answer/9888077?hl=en))   , its all about app review by the playstore officials, not the app rating/review by the app users.   the app by [sakibanda](https://github.com/sakibanda) is suspended, bacause playstore suspected that something was hidden from the app reviewer.  Please correct me if I am wrong.

**Comment by SuddenH4X** [comment]
@suhasrekhu Thanks for pointing this out. I think you are correct - there is a clear distinction between "evading app reviews" (Google's internal process) and managing "user ratings/reviews".

As the library does not attempt to hide code or behavior from the Play Store reviewers, it does not violate this section of the policy. I will close this issue for now. Should new concerns arise, feel free to open a new ticket.
